### PR TITLE
MGMT-16704: Add organization capability to control soft timeouts

### PR DESCRIPTION
**Pull request by ori-amizur**
We would like to control invocation of soft timeouts based on organizational permissions.  Therefore, capability for allowing soft timeouts was added.

This capability is used in conjunction with corresponding soft timeouts environment variable.  Soft timeouts will be used for cluster insallation if both it is enabled by the assisted service environment variable and if it is allowed by capability.

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md
/cc @gamli75 
/cc @jhernand

### CODE REVIEW

File: internal/bminventory/inventory_test.go [hunk 1]
```diff
@@ -15060,6 +15060,56 @@ var _ = Describe("TestRegisterCluster", func() {
...
+					OpenshiftVersion: swag.String("4.12.0-someFakeFlavour"),
+					CPUArchitecture:  common.X86CPUArchitecture,
+					PullSecret:       swag.String(fakePullSecret),
+				},
+			})
+			Expect(reply).To(BeAssignableToTypeOf(installer.NewV2RegisterClusterCreated()))
+			actual := reply.(*installer.V2RegisterClusterCreated)
+			Expect(actual.Payload.OrgSoftTimeoutsEnabled).To(Equal(softTimeouts))
+		},
+			Entry("soft timeouts prohibited", false),
```
Comment (tsorya) [inline internal/bminventory/inventory_test.go:15108]: Maybe we should test that if there is an error when we ask for capability we don't fail?

Comment (ori-amizur) [reply internal/bminventory/inventory_test.go:15108]: What will we do in case of failure?

Comment (tsorya) [reply internal/bminventory/inventory_test.go:15108]: continue, this is what is done in code, no?

Comment (ori-amizur) [reply internal/bminventory/inventory_test.go:15108]: Yes, so what is the point to catch error?

Comment (ori-amizur) [reply internal/bminventory/inventory_test.go:15108]: I will add error handling

**Pull request merged**